protocol-spec/enumeration.md

```diff
@@ -244,5 +351,13 @@ None. Server-to-Client message only.
 ]
 ```
 
+### MessageAttributes
+
+**Description:** An collection of message attributes \(this object is always the child of a Device Message type name within a DeviceList or DeviceAdded message\). Not all attributes are relevant for all Device Messages on all Devices; in these cases the attributes will not be included.
```

Comment:
Fixing the first 2 points.
Personally I prefer the look of:

```json
"StopDeviceCmd": {}
```

Over:

```json
"StopDeviceCmd": null
```

The latter has the feel that the command shouldn't be in the list instead of not having any attributes.

Reply:
Oh. Yeah. Sorry, I meant {}, null object was a bit ambiguous heh.

Reply:
So that's how it currently works (in the C# this is represented by the attributes being set to null). I'll see if I can clarify this at the spec level.